### PR TITLE
Reshape Unpooling Output

### PR DESCRIPTION
**Pull request by mcr229**
Reshape Unpooling output

### CODE REVIEW

**Comment by mcr229** [comment]
cc: @digantdesai @fbarchard @alankelly @gonnet  for reviews

File: src/subgraph/unpooling-2d.c [hunk 1]
```diff
@@ -57,17 +57,44 @@ static enum xnn_status reshape_unpooling_operator(
...
   const size_t input_width = values[input_id].shape.dim[2];
-  return xnn_reshape_unpooling2d_nhwc_x32(
+  const size_t channel_dim = values[input_id].shape.dim[3];
+
+  struct xnn_value* output_value = values + output_id;
+  enum xnn_status status = xnn_status_invalid_state;
+  const size_t old_workspace_size = opdata->workspace_size;
+  size_t output_height, output_width;
+
+  status =  xnn_reshape_unpooling2d_nhwc_x32(
```
Comment (fbarchard) [inline src/subgraph/unpooling-2d.c:74]: return xnn_reshape_unpooling2d_nhwc_x32(

File: test/unpooling-2d.cc [hunk 2]
```diff
@@ -222,3 +222,84 @@ TEST_F(Unpooling2DTestX32, matches_operator_api)
...
+
+  // Call subgraph API.
+  xnn_subgraph_t subgraph = nullptr;
+  ASSERT_EQ(xnn_status_success, xnn_create_subgraph(2, /*flags=*/0, &subgraph));
+  std::unique_ptr<xnn_subgraph, decltype(&xnn_delete_subgraph)> auto_subgraph(subgraph, xnn_delete_subgraph);
+
+  uint32_t input_value_id = XNN_INVALID_NODE_ID;
+  ASSERT_EQ(
+    xnn_status_success, xnn_define_tensor_value(
+                          subgraph, xnn_datatype_fp32, input_value_dims.size(), input_value_dims.data(), nullptr,
```
Comment (fbarchard) [inline test/unpooling-2d.cc:238]: wrong amount of indent.  either a long line or indent by 2

Comment (mcr229) [reply test/unpooling-2d.cc:238]: hmm i see, I'm just keeping consistent with the tests above, but let me know if you want to fix all of those in this PR.

File: src/subgraph/unpooling-2d.c (hunk 1 above)
Comment (mcr229) [reply src/subgraph/unpooling-2d.c]: I resolved the extra space there. Not sure If i can return early here though

**Pull request merged**